File: array and string/longest_common_prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs)==0:
            return ""
        small = min(strs)
        s = ""
        
        for i in range(len(small)):
            flag = True
            c = small[i]
            
            for j in range(len(strs)):
                if strs[j][i]!=c:
                    flag = False
                    break
            if flag == False:
                break
            else:
                s = s + c
        return s
                

File: array and string/longest_common_prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        
        if(strs.length==0)
        {
            return "";
        }
        String small = strs[0];
        String str = "";
        
        for(int i=0; i<strs.length; i++)
        {
            if(strs[i].length()<small.length())
            {
                small = strs[i];
            }
        }
        
        for(int i=0; i<small.length(); i++)
        {
            boolean flag = true;
            char c = small.charAt(i);
            
            for(int j=0; j<strs.length; j++)
            {
                if(strs[j].charAt(i)!=c)
                {
                    flag = false;
                    break;
                }
            }
            
            if (flag==false)
            {
                break;
            }
            else
            {
                str = str + c;
            }
        }
        
        return str;
        
    }
}